### PR TITLE
Private key sweeping

### PR DESCRIPTION
**Pull request by zeptin**
https://dev.azure.com/Stratisplatformuk/STRAX/_boards/board/t/STRAX%20Team/Stories/?workitem=5203

Here is an example of a swept transaction successfully broadcast:

https://chainz.cryptoid.info/strax-test/block.dws?46039.htm

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/BalancesController.cs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.SmartContracts.CLR;
+using Stratis.SmartContracts.Core.Util;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.ReflectionExecutor.Controllers
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/BalancesController.cs:11]: How come this is under smart contracts's set of controllers? 

Perhaps we can also put this with the other address balance controller methods, i.e. in the blockstore controller.

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/BalancesController.cs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            var coinView = new CoinviewContext();
+
+            const int batchSize = 1000;
+            IEnumerable<ChainedHeader> allBlockHeaders = this.chainIndexer.EnumerateToTip(this.chainIndexer.Genesis);
+
+            int totalBlocksCounted = 0;
+
+            int i = 0;
+
+            while (true)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/BalancesController.cs:77]: This logic is obvs VERY similar to the `UtxoIndexer` above. Was this intentional or can this be consolidated?

**Review by fassadlr — APPROVED** [review]
Looks good, just 2 comments 👍

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/BalancesController.cs (hunk 1 above)
Comment (zeptin) [reply src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/BalancesController.cs:11]: Basically it's just bringing in this PR from the old 3.0.9.0 branch:

https://github.com/stratisproject/StratisBitcoinFullNode/pull/4186

**Pull request merged**